dev/Deployment/DeploymentManager.cpp

```diff
@@ -414,8 +414,8 @@ namespace winrt::Microsoft::Windows::ApplicationModel::WindowsAppRuntime::implem
                 deploymentResult.ActivityId());
         }
 
-        return !deploymentOperationHResult ? deploymentOperationHResult :
-            (deploymentOperationExtendedHResult ? deploymentOperationExtendedHResult : deploymentOperationHResult);
+        return FAILED(deploymentOperationHResult) ? deploymentOperationHResult :
```

Comment:
just double-checking - this is a functional change (and I believe a bug fix).

Reply:
For context, here's the original error:
```
##[error]18. PREfast Error C6230 - File: file:///C:/__w/1/s/dev/Deployment/DeploymentManager.cpp. Line: 418. Column 14. 
Tool: PREfast: Rule: C6230 ((Unknown Rule Name)). 
Implicit cast between semantically different integer types.
```
My understanding of the intention of the code author is:
- if deploymentOperationHResult contains an error then just return it and be done
- otherwise, if deploymentOperationExtendedHResult contains an error then just return it be done
- otherwise, return deploymentOperationHResult which should have contained "success"

I think PREfast does not like the use of HRESULT types deploymentOperationHResult and deploymentOperationExtendedHResult where booleans are expected in the logical expression, so I added FAILED() to do the conversion. Looks like the original code can handle S_OK (i.e., 0) being returned in deploymentOperationHResult and deploymentOperationExtendedHResult fine, but S_FALSE (i.e., 1) would not be handled correctly, but in reality this probably moot, because I don't recall the Package Manager ever using S_FALSE.

Reply:
Assuming the "!" near the front is intended probably minimizes the risk. Updated the fix and added comment for a possible interpretation. Thanks!